### PR TITLE
remove modal for user commonality

### PR DESCRIPTION
**Pull request by reteps**
Courses in Common now automatically load below the users name!
image

### CODE REVIEW

File: js/user.js [hunk 1]
```diff
@@ -1,24 +1,21 @@
...
-        button.prepend(img);
-        button.querySelector("input").style.paddingLeft = "4px";
-        button.style.cursor = "pointer";
-        button.addEventListener("click", () => {
-            if (loadCommonCourses == null) {
-                loadCommonCourses = getCoursesInCommon(document.location.href.match(/\/(\d+)\//)[1]);
-            }
-            openModal("user-courses-in-common-modal", { loadCommonCourses: loadCommonCourses, userId: document.location.href.match(/\/(\d+)\//)[1] });
-        });
+    const userID = document.location.href.match(/\/(\d+)\//)[1]
```
Comment (atomicproton) [inline js/user.js:5]: I'd end each statement with a semicolon just to match the style of the rest of the code. 

Also in my experience I've run in to many strange bugs when dealing with semicolon auto-insertion.

**Review by aopell — CHANGES_REQUESTED** [review]
Yes, please add semicolons!

**Comment by reteps** [comment]
Alright! I have not experienced that issue before, but I added them in.

**Comment by reteps** [comment]
1 More change, now click on the gradeText to edit instead of the icon.

**Comment by glen3b** [comment]
Disclaimer: I haven't looked at the code and I'm not currently in a position to test it.

Marking here some things we'll want to verify and/or address:
- Sometimes profiles aren't blank. I think a permissions thing on the LAUSD instance makes them so. We'll want to make sure your change doesn't cause unexpected behavior either with non-blank profiles or on the user's own profile.
- If you can keep one change per PR that would be appreciated. Please move the gradeText change to another PR if you can. (We can handle keeping the commit history clean when it comes to that.)

From a design perspective I'm definitely not opposed to inlining the courses in common on profile pages, assuming there's not a large performance impact. Thanks for taking a look at this!

**Comment by reteps** [comment]
This does work with non-blank profiles, e.g. my own profile.
image

**Comment by reteps** [comment]
Grade text has been moved to a seperature pull request.

File: js/user.js [hunk 2]
```diff
@@ -1,24 +1,16 @@
...
+    const userID = document.location.href.match(/\/(\d+)\//)[1];
+    const loadCommonCourses = getCoursesInCommon(userID);
+    const container = createElement("div", ["splus-modal-contents"], {}, [
+                        createElement("ul", ["setting-entry", "common-realm-list"], { id: "user-courses-in-common-list" }, [])
+                    ]);
+    const title = createElement("h4", [], {textContent: "Courses In Common"});
+    page.appendChild(title);
+    page.appendChild(container);
+    setCourseListModalContent({loadCommonCourses });
 })();
```
Comment (glen3b) [inline js/user.js:13]: I think for anonymous async functions like this one we like to have an error handler or some other way to make sure exceptions don't get swallowed. I believe grades.js has one if you want to model off of it. In lieu of a larger refactoring, simply logging the error should be fine.

File: js/user.js [hunk 3]
```diff
@@ -1,24 +1,16 @@
...
-    }
+(async function () {
+    const page = document.getElementById("main-inner");
+    if (!page) return;
+    const userID = document.location.href.match(/\/(\d+)\//)[1];
+    const loadCommonCourses = getCoursesInCommon(userID);
+    const container = createElement("div", ["splus-modal-contents"], {}, [
+                        createElement("ul", ["setting-entry", "common-realm-list"], { id: "user-courses-in-common-list" }, [])
+                    ]);
+    const title = createElement("h4", [], {textContent: "Courses In Common"});
```
Comment (glen3b) [inline js/user.js:9]: Stylistically, can we make this closer to a "proper" section? See this screenshot (Firefox):
image
Can we have "Courses In Common" be closer stylistically to "About Me" or "Contact Information"?

**Review by glen3b — CHANGES_REQUESTED** [review]
Thanks again for looking at this. There's a couple small changes I'd like to see addressed. The one thing I didn't point out there is that I'm currently having functionality issues when I pull this up on my own profile on the main schoology.com instance, and I'm not currently in a position to troubleshoot that.

Other than that, looks like a nice change! As a recommendation, for future changes, I've found when making PRs, keeping your `develop` branch clean and up-to-date with ours, and forking off your `develop` branch into feature branches on your own repository for making PRs helps avoid PR cross-contamination and makes including upstream changes simple.

**Comment by reteps** [comment]
closing, I can't figure out reverting this correctly so I am redoing this.

**Pull request closed without merging**